libs/libc/stdio/lib_ungetc.c

```diff
@@ -43,19 +43,17 @@ int ungetc(int c, FAR FILE *stream)
   int nungotten;
 #endif
 
-  /* Verify that a non-NULL stream was provided */
+  /* Verify that a non-NULL stream was provided and c is not EOF */
 
-  if (!stream)
+  if (!stream || c == EOF)
```

Comment:
Initially I was thinking it couldn't never test 'c', but the only way it happen is if 'stream' is NULL, so the everything is fine!